paimon-core/src/main/java/org/apache/paimon/catalog/Catalog.java

```diff
@@ -273,7 +274,15 @@ class TableNotExistException extends Exception {
 
         private static final String MSG = "Table %s does not exist.";
 
-        private final Identifier identifier;
+        private Identifier identifier;
+
+        public TableNotExistException(Path tablePath) {
```

Comment:
Can we just introduce an unknown database for this?

Reply:
done.